controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -218,11 +233,82 @@ func (r *StorageConsumerReconciler) setCephResourceStatus(name string, kind stri
 	)
 }
 
+func (r *StorageConsumerReconciler) deletionPhase() error {
+	lastHeartBeat := r.storageConsumer.Status.LastHeartbeat
+	if !lastHeartBeat.IsZero() && time.Since(lastHeartBeat.Time) >= time.Duration(5*time.Minute) {
+		if err := r.annotateCephResources(); err != nil {
+			return err
+		}
+	}
+
+	if controllerutil.RemoveFinalizer(r.storageConsumer, storageConsumerFinalizer) {
+		if err := r.update(r.storageConsumer); err != nil {
+			return fmt.Errorf("failed to remove finalizer: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (r *StorageConsumerReconciler) annotateCephResources() error {
+
+	scrList := &ocsv1alpha1.StorageClassRequestList{}
+	if err := r.list(scrList,
+		client.InNamespace(r.storageConsumer.Namespace),
+		client.MatchingFields{ConsumerUUIDLabel: string(r.storageConsumer.UID)}); err != nil {
+		return fmt.Errorf("failed to list storageclassrequests that are referring storageconsumer: %v", err)
+	}
+	for idx := range scrList.Items {
+		scr := &scrList.Items[idx]
+		for idx := range scr.Status.CephResources {
+			resource := scr.Status.CephResources[idx]
+			switch resource.Kind {
+			case "CephBlockPoolRadosNamespace":
+				rns := &rookCephv1.CephBlockPoolRadosNamespace{}
+				rns.Name = resource.Name
+				rns.Namespace = r.storageConsumer.Namespace
+				if err := r.get(rns); err != nil {
+					return fmt.Errorf("failed to get CephBlockPoolRadosNamespace: %v", err)
```

Comment:
should handle not found error as well?

Reply:
reimplemented a straight forward way, pls re-review. thanks.

Reply:
resolving after confirming offline.